product_mrp_info/tests/test_product_mrp_info.py

```diff
@@ -0,0 +1,92 @@
+        cls.mo_obj = cls.env["mrp.production"]
+        cls.bom_obj = cls.env["mrp.bom"]
+        cls.boml_obj = cls.env["mrp.bom.line"]
+
+        cls.manufacture_route = cls.env.ref("mrp.route_warehouse0_manufacture")
+
+        cls.attribute = cls.env["product.attribute"].create(
+            {"name": "Test Attribute", "create_variant": "always"}
+        )
+        cls.value1 = cls.env["product.attribute.value"].create(
+            {"name": "Value 1", "attribute_id": cls.attribute.id}
+        )
+        cls.value2 = cls.env["product.attribute.value"].create(
+            {"name": "Value 2", "attribute_id": cls.attribute.id}
+        )
+        cls.product = cls.pt_obj.create(
+            {
+                "name": "Test Template",
+                "attribute_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "attribute_id": cls.attribute.id,
+                            "value_ids": [(6, 0, [cls.value1.id, cls.value2.id])],
+                        },
+                    )
+                ],
+                "route_ids": [(6, 0, cls.manufacture_route.ids)],
+            }
+        )
+        cls.variant_1 = cls.product.product_variant_ids[0]
+        cls.variant_2 = cls.product.product_variant_ids[1]
+        cls.bom = cls.bom_obj.create(
+            {"product_tmpl_id": cls.product.id, "product_qty": 1.0}
+        )
+        # Create 3 MO's
+        cls.mo_1 = cls.mo_obj.create(
+            {
+                "name": "MO ABC",
+                "product_id": cls.variant_1.id,
+                "product_uom_id": cls.variant_1.uom_id.id,
+                "product_qty": 2,
+                "bom_id": cls.bom.id,
+            }
+        )
+        cls.mo_2 = cls.mo_obj.create(
+            {
+                "name": "MO XYZ",
+                "product_id": cls.variant_1.id,
+                "product_uom_id": cls.variant_1.uom_id.id,
+                "product_qty": 3,
+                "bom_id": cls.bom.id,
+            }
+        )
+        cls.mo_3 = cls.mo_obj.create(
+            {
+                "name": "MO QWE",
+                "product_id": cls.variant_2.id,
+                "product_uom_id": cls.variant_2.uom_id.id,
+                "product_qty": 6,
+                "bom_id": cls.bom.id,
+            }
+        )
+
+    def test_01_mo_counters(cls):
+        cls.assertEqual(len(cls.product.product_variant_ids), 2)
+        cls.assertEqual(cls.product.mo_count, 3)
+        cls.assertEqual(cls.variant_1.mo_count, 2)
+        cls.assertEqual(cls.variant_2.mo_count, 1)
+
+    def test_02_actions(cls):
+        # template
+        res = cls.product.action_view_mrp_productions()
+        count = cls.mo_obj.search_count(res["domain"])
+        cls.assertEqual(count, 3)
+        # variant
+        res = cls.variant_1.action_view_mrp_productions()
+        count = cls.mo_obj.search_count(res["domain"])
+        cls.assertEqual(count, 2)
```

Comment:
You do not have to change `self -> cls` in this part of the tests. This is what is causing to fail the pre-commit check.

Reply:
During replacing self with cls for setupclass, it was wrongly replaced in tests as well. Modified now